analytic_hours_block/hours_block.py

```diff
@@ -47,10 +47,14 @@ def _compute_hours(self, cr, uid, ids, fields, args, context=None):
         for block in self.browse(cr, uid, ids, context=context):
             result[block.id] = {'amount_hours_block': 0.0,
                                 'amount_hours_block_done': 0.0}
+            block_analytic_id = block.account_analytic_id.id
             # Compute hours bought
             for line in block.invoice_id.invoice_line:
                 hours_bought = 0.0
-                if line.product_id and line.product_id.is_in_hours_block:
+                line_account_analytic_id = line.account_analytic_id.id
+                if line.product_id and \
+                        line.product_id.is_in_hours_block and \
+                        block_analytic_id == line_account_analytic_id:
```

Comment:
`\` for continuation lines are to be avoided, parentheses should be preferred. Example:

```python
if (line.product_id and
        line.product_id.is_in_hours_block and
        block_analytic_id == line_account_analytic_id):
```